shutdown.go

```diff
@@ -0,0 +1,65 @@
+package gokit
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+// Server will create a http.Server from the Go standard library
+type Server struct {
```

Comment:
Does it make sense to the server to embed the `http.Server` struct instead?

We wouldn't have to define these data fields, but we have to take more care to implement the functions, otherwise they'll end up calling the `http.Server` struct's methods instead.

Reply:
It's because we don't own the `http.Server` code, so we cannot control how it will be changing in the future. For example,
- We may have a method name that doesn't clash now, but clash in the future, or
- We may have a variable like `Timeout` we're using now that the standard library decides to use in the future.

To ensure that this code will not have problems without us checking every version upgrade is to not embed it. The disadvantage is the setup code at the beginning of `ListenAndServe()` for every additional variable we need.

Reply:
That makes sense :+1: